### DISCUSSION TITLE
What kind of protection has ExcelJS?

### DISCUSSION THREAD

**Discussion by ayushkumar9333** [exceljs/exceljs#1686] [Q&A]
[Q] how to protect workwook. i am using worksheet.protect('test123', {}) but its giving me partial protection. i want to use workbook protection to stop user from adding sheets from below add sheet sign. i am using angular6 to create excel file.

**Comment by mah110020** [comment]
Functionality for preventing the adding/deleting/renaming of worksheets, [does exist in Excel](https://www.extendoffice.com/documents/excel/1772-excel-prevent-insert-worksheet.html) ("Protect Workbook"), but does not appear to be implemented in ExcelJS. This library currently only supports protecting the worksheet with permissions shown in the [Sheet Protection Options](https://github.com/exceljs/exceljs#sheet-protection) table.

Perhaps you can make a feature request? I would also find this functionality useful.

**Comment by damingerdai** [comment]
it looks like nobody want to make a feature request

**Reply by Siemienik** [reply]
There is some people like me which would like as long as there will find others which want to pay me for doing that.
